rai/core_test/rpc.cpp

```diff
@@ -962,6 +962,28 @@ TEST (rpc, history)
 	ASSERT_EQ (rai::test_genesis_key.pub.to_account (), std::get<1> (history_l[4]));
 	ASSERT_EQ (rai::genesis_amount.convert_to<std::string> (), std::get<2> (history_l[4]));
 	ASSERT_EQ (genesis.hash ().to_string (), std::get<3> (history_l[4]));
+	//Tests filtering
+	auto account2 (system.wallet (0)->deterministic_insert ());
+	auto send2 (system.wallet (0)->send_action (rai::test_genesis_key.pub, account2, system.nodes[0]->config.receive_minimum.number ()));
+	ASSERT_NE (nullptr, send2);
+	auto receive2 (system.wallet (0)->receive_action (static_cast<rai::send_block &> (*send2), account2, system.nodes[0]->config.receive_minimum.number ()));
+	ASSERT_NE (nullptr, receive2);
+	boost::property_tree::ptree request2;
+	request2.put ("action", "history");
+	request2.put ("account", rai::test_genesis_key.pub.to_account ());
+	boost::property_tree::ptree other_account;
+	other_account.put ("", account2.to_account ());
+	boost::property_tree::ptree filtered_accounts;
+	filtered_accounts.push_back (std::make_pair ("", other_account));
+	request2.add_child ("account_filter", filtered_accounts);
+	request2.put ("count", 100);
+	test_response response2 (request2, rpc, system.service);
+	while (response2.status == 0)
+	{
+		system.poll ();
+	}
+	auto history_node2 (response2.json.get_child ("history"));
```

Comment:
currently this returns an error rather than a history node. I think you will find that using account_history rather than history as your RPC call will work better as history has been deprecated and does not function exactly the same as account_history would.